src/System.Memory/tests/Performance/Perf.ReadOnlySequence.TryGet.cs

```diff
@@ -0,0 +1,374 @@
+                        while (buffer.TryGet(ref p, out ReadOnlyMemory<char> memory))
+                            localInt ^= memory.Length;
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void String(int size, int start, int lenght)
+        {
+            ReadOnlyMemory<char> strMemory = new string('a', size).AsMemory();
+            strMemory = strMemory.Slice(start, lenght);
+            var buffer = new ReadOnlySequence<char>(strMemory);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < size / 10; j++)
+                        {
+                            var p = buffer.Start;
+                            while (buffer.TryGet(ref p, out ReadOnlyMemory<char> memory))
+                                localInt ^= memory.Length;
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Char_Empty()
+        {
+            ReadOnlySequence<char> buffer = ReadOnlySequence<char>.Empty;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < Benchmark.InnerIterationCount; j++)
+                        {
+                            var p = buffer.Start;
+                            while (buffer.TryGet(ref p, out ReadOnlyMemory<char> memory))
+                                localInt ^= memory.Length;
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Char_Default()
+        {
+            ReadOnlySequence<char> buffer = default;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < Benchmark.InnerIterationCount; j++)
```

Comment:
As @pakrym mentioned, we shouldn't have two loops going over InnerIterationCount (here and elsewhere)